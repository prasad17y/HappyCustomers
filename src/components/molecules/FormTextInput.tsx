import React from 'react';
import {View, Text, StyleSheet, TextInputProps} from 'react-native';
import StyledTextInput from '../atoms/StyledTextInput';

interface FormTextInputProps extends TextInputProps {
  error?: string;
  touched?: boolean;
}

const FormTextInput: React.FC<FormTextInputProps> = ({
  error,
  touched,
  ...props
}) => {
  const hasError = !!(error && touched);
  return (
    <View style={styles.fieldContainer}>
      <StyledTextInput hasError={hasError} {...props} />
      <View style={styles.errorContainer}>
        {hasError && <Text style={styles.errorText}>{error}</Text>}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  fieldContainer: {
    marginBottom: 8,
  },
  errorContainer: {
    height: 20,
    justifyContent: 'center',
  },
  errorText: {
    color: 'red',
  },
});

export default React.memo(FormTextInput);
