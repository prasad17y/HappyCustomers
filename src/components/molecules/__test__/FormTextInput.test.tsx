import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react-native';
import FormTextInput from '../FormTextInput';

describe('FormTextInput', () => {
  it('should render the text input correctly', () => {
    render(<FormTextInput value="test" onChangeText={jest.fn()} />);
    expect(screen.getByDisplayValue('test')).toBeTruthy();
  });

  it('should not display an error message if not touched', () => {
    render(
      <FormTextInput
        value=""
        onChangeText={jest.fn()}
        error="This is an error"
        touched={false}
      />,
    );
    expect(screen.queryByText('This is an error')).toBeNull();
  });

  it('should display an error message when touched and an error exists', () => {
    render(
      <FormTextInput
        value=""
        onChangeText={jest.fn()}
        error="This is an error"
        touched={true}
      />,
    );
    expect(screen.getByText('This is an error')).toBeTruthy();
  });

  it('should match the snapshot', () => {
    const tree = render(
      <FormTextInput value="snapshot" onChangeText={jest.fn()} />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
