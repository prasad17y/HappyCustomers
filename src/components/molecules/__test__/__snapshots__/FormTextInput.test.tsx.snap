// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTextInput should match the snapshot 1`] = `
<View
  style={
    {
      "marginBottom": 8,
    }
  }
>
  <View
    style={
      [
        {
          "borderBottomColor": "#e0e0e0",
          "borderBottomWidth": 1,
        },
        false,
      ]
    }
  >
    <TextInput
      onChangeText={[MockFunction]}
      placeholderTextColor="#a0a0a0"
      style={
        [
          {
            "color": "#191a1a",
            "fontSize": 16,
            "paddingVertical": 12,
          },
          undefined,
        ]
      }
      value="snapshot"
    />
  </View>
  <View
    style={
      {
        "height": 20,
        "justifyContent": "center",
      }
    }
  />
</View>
`;
